.github/workflows/task-rust-code-review.yml

```diff
@@ -0,0 +1,184 @@
+name: Rust Code Review
+
+on:
+  workflow_call:
+    inputs:
+      pr_number:
+        required: true
+        type: number
+        description: "The PR number to review"
+      repository:
+        required: true
+        type: string
+        description: "The repository name with owner (e.g., owner/repo)"
+
+jobs:
+  review-rust-code:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install requests openai
+
+      - name: Get changed Rust files
+        id: changed-files
```

Comment:
there's https://github.com/marketplace/actions/changed-files too though this doesn't look too complicated